File: pinyougou-interface/src/main/java/cn/itcast/core/service/GoodsService.java
package cn.itcast.core.service;

import cn.itcast.core.pojo.good.Goods;
import entity.PageResult;
import entity.Result;
import pojogroup.GoodsVo;

public interface GoodsService {
    //新增商品
    void add(GoodsVo vo);
//  Goods表查询:分页查询+条件查询
    PageResult search(Integer page, Integer rows, Goods goods);

    //商品修改之 信息回显 根据id查询商品
    GoodsVo findOne(Long id);
//    商品修改按钮触发,信息回显修改之后,的保存操作
    void update(GoodsVo vo);

//    商品审核:删除  支持批量
    void delete(Long[] ids);
//    商品审核:同意/不同意  支持批量
    void updateStatus(Long[] ids, String status);
}

File: pinyougou-web-manager/src/main/webapp/js/service/goodsService.js
//服务层
app.service('goodsService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../goods/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../goods/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../goods/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../goods/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../goods/update.do',entity );
	}

	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../goods/search.do?page='+page+"&rows="+rows, searchEntity);
	}

    //删除:商品审核页:功能:goods.html  ids:复选框批量id信息   status:用户绝对审批的状态码
    this.dele=function(ids){
        return $http.get('../goods/delete.do?ids='+ids);
    }
	//审批:商品审核页:功能:goods.html  ids:复选框批量id信息   status:用户绝对审批的状态码
	this.updateStatus = function(ids,status){
		return $http.get('../goods/updateStatus.do?ids='+ids+"&status="+status);
	}
});
